### ISSUE TITLE
Should a name change of a focused control (such as button) be conveyed?

### ISSUE THREAD

**Issue by mfairchild365** [nvaccess/nvda#11224]
### Steps to reproduce:

1. Using NVDA and Chrome, navigate to https://js-pce2dd.stackblitz.io
2. Navigate to each button and activate it
3. Observe that NVDA does not announce name changes for example 2, but it does for example 1.

Note that the name change is conveyed if the `aria-label` is updated (example 2), but not if the inner text is updated (example 1).

### Actual behavior:

Nothing is conveyed after activating the button in example 2.

### Expected behavior:

I expect to hear the new name.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2020.1

#### Windows version:
Windows 10

#### Name and version of other software in use when reproducing the issue:
Chrome 83 and Firefox 77 

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
I have not

#### If addons are disabled, is your problem still occuring?
Yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

**Comment by Adriani90** [comment]
cc: @jcsteh at least for Firefox.

**Comment by jcsteh** [comment]
The name change should ideally be conveyed, yes. The reason it isn't for the text content case is that neither browser fires a nameChange event in that case. Fixing that problem is a lot harder than it might seem, especially without hurting performance. The Mozilla bug for is this is https://bugzilla.mozilla.org/show_bug.cgi?id=1422913

**Comment by bhavyashah** [comment]
Assuming that this issue would be automatically resolved if the concerned Mozilla bug is fixed, is it alright to close this ticket with a NeedsExternalFix label? If my assumption is wrong, then never mind this question.

**Comment by Adriani90** [comment]
@bhavyashah I would not close this issue, we can track here the Mozilla bug and after that we can close this.

**Comment by Adriani90** [comment]
It seems in Firefox and Chromium all buttons are reported by NVDA. In chrome though, the first button in example 1 is reported only first time when clicking on it, afterwards is not reported anymore.
Examples 3 and 4 are repeated twice or three times in both Firefox and Chromium. See also related issue #7996.

@mfairchild365 can you please provide the exact code of the examples? I cannot see it in the dev console.

**Comment by codeofdusk** [comment]
I recently opened #20018 (which is a Chromium-specific variant of this pattern).
